### ISSUE TITLE
GoDef fails if file has unsaved changes

### ISSUE THREAD

**Issue by cmcginty** [fatih/vim-go#1531]
### What did you do? (required. The issue will be **closed** when not provided.)

Call :GoDef when cursor on a type. The go files was modified by adding a trailing space to the end of one line, but not saved.

### What did you expect to happen?

Go to the def.

### What happened instead?

Vim Go reports error, "vim-go: godef: no declaration found for XXXXX"
If I save the file, and then call :GoDef, the declaration is found immediately. 

### Configuration

* vim version (major, minor, and included patches from `:version`): version 8.0.1175, Included patches: 1-1175

* vim-go version: 011dcad
* go version (`go version`): go version go1.9.1 darwin/amd64
* operating system: Mac OS Sierra
* `vimrc` you used to reproduce: see [cmcginty/dotfiles/vim](https://github.com/cmcginty/dotfiles/tree/master/vim)

**Comment by Carpetsmoker** [comment]
Did you add the `import` line for the package? e.g. this will fail:

```
package main

func main() {
    fmt.Println()
}
```

but this works:

package main

import "fmt"

func main() {
    fmt.Println()
}
```

You can use `:GoImport fmt` to add the import.

**Comment by Carpetsmoker** [comment]
I'll close this for now due to lack of feedback. Feel free to comment if you have further problems or questions and we can reopen this issue.

**Comment by cmcginty** [comment]
Yes, the file had the correct syntax. I think that was pretty clear from the report.

`The go files was modified by adding a trailing space to the end of one line, but not saved.`
`If I save the file, and then call :GoDef, the declaration is found immediately.`

**Comment by bhcleek** [comment]
@cmcginty I tried reproducing what you've described by putting a space at the end of the `fmt.Println` of a file with this go code:
```
package main

import "fmt"

func main() {
	fmt.Println("Hello World")
}
```

But it works fine for me using either on latest master and `011dcad`. I've tried it with `g:go_def_mode='guru'` and `g:go_def_mode='godef'`.

Have you run `:GoUpdateBinaries` recently? Is it possible you're using an old version of guru or godef?
